### PR TITLE
[Artists] Respond with json for name lookup & deletion

### LINKED ISSUE #503: Getting a nonexistent artist by name redirects when using JSON
Though not necessarily the intended method of getting an artist by name in the api, `/artsts/[name].json` works. While this works, it redirects to `/artists/show_or_new` if an artist by that name doesn't exist, regardless of response type.
https://github.com/e621ng/e621ng/blob/179bb69f1da33de904cc9c0788e4db3a2fc5ac54/app/controllers/artists_controller.rb#L26-L38

### PR DESCRIPTION
**Pull request by DonovanDMC**
Fixes #503

This makes `GET /artists/{name}` return a proper 404 json response as it should, as well as returning a json response when deleting an artist.

### CODE REVIEW

File: app/controllers/artists_controller.rb [hunk 1]
```diff
@@ -50,11 +57,18 @@ def update
   end
 
   def destroy
-    if !@artist.deletable_by?(CurrentUser.user)
+    unless @artist.deletable_by?(CurrentUser.user)
       raise User::PrivilegeError
     end
     @artist.update_attribute(:is_active, false)
-    redirect_to(artist_path(@artist), :notice => "Artist deleted")
+    respond_to do |format|
```
Comment (Earlopain) [inline app/controllers/artists_controller.rb]:
```suggestion
    respond_with(@artist) do |format|
      format.html do
        redirect_to(artist_path(@artist), notice: "Artist deleted")
      end
    end
```

**Comment by DonovanDMC** [comment]
Thanks github

**Comment by Earlopain** [comment]
> Thanks github

It's my fault really, for the less than ideal test setup. I did f722dbb4c38c8a8aed06d3948149b8f990a1cb1e which hopefully improves on that a bit for future PRs. Didn't fail immediatly so maybe it actually works.

**Comment by DonovanDMC** [comment]
> > Thanks github
> 
> It's my fault really, for the less than ideal test setup. I did [f722dbb](https://github.com/e621ng/e621ng/commit/f722dbb4c38c8a8aed06d3948149b8f990a1cb1e) which hopefully improves on that a bit for future PRs. Didn't fail immediatly so maybe it actually works.

I was more referring to what happened in d1013724e57eaf982557044250a830dc9afba9e5 (before I force pushed it out) when I commited that suggestion, but it seems like that was actually your doing, since your suggestion added on and didn't remove the rest 🙃

The test fails are annoying yes, but I've more or less learned to accept them as something that will happen

**Pull request merged**